scripts/test-e2e-full-stack.sh

```
@@ -832,17 +868,39 @@ copy_skill_fixture() {
+    local stage_parent stage_dir
     ensure_directory_writable "$dest_root" "copying skill fixture"
-    mkdir -p "$dest_root/$dest_name"
-    cp -R "$fixture_dir"/. "$dest_root/$dest_name/"
+    stage_parent=$(dirname "$dest_root")
+    stage_dir=$(mktemp -d "$stage_parent/.defenseclaw-skill-fixture.XXXXXX" 2>/dev/null || mktemp -d "${TMPDIR:-/tmp}/defenseclaw-skill-fixture.XXXXXX")
+    if ! cp -R "$fixture_dir"/. "$stage_dir/"; then
+        rm -rf "$stage_dir"
+        return 1
+    fi
+    chmod -R u+rwX,go+rX "$stage_dir" 2>/dev/null || true
+    rm -rf "$dest_root/$dest_name" 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)